src/functions.php

```diff
@@ -78,6 +78,7 @@ function stream_for($resource = '', array $options = [])
 {
     switch (gettype($resource)) {
         case 'string':
+        case 'integer':
```

Comment:
Can you change this to `is_scalar` and remove the case for "string"? That would support casting any PHP scalar type to a string.